### PR TITLE
feat: ORV2-1702 No challenge login claim company

### PR DESCRIPTION
**Pull request by krishnan-aot**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-853-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-853-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-853-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-853-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/common/authentication/LoginRedirect.tsx [hunk 1]
```diff
@@ -37,17 +37,38 @@ export const LoginRedirect = () => {
...
+            associatedCompanies,
+            pendingCompanies,
+            migratedTPSClient,
+            user,
+          } = userContextData;
           // If the user does not exist
           if (!user?.userGUID) {
             // The user is in pending companies => Redirect them to User Info Page.
+            // i.e., The user has been invited.
             if (pendingCompanies.length > 0) {
```
Comment (erikataot) [inline frontend/src/common/authentication/LoginRedirect.tsx]:
```suggestion
            if (pendingCompanies?.length > 0) {
```

File: frontend/src/common/authentication/LoginRedirect.tsx [hunk 2]
```diff
@@ -37,17 +37,38 @@ export const LoginRedirect = () => {
...
               navigate(CREATE_PROFILE_WIZARD_ROUTES.WELCOME);
             }
-            // The user and company does not exist => Redirect them to Add new company page.
-            else if (associatedCompanies.length < 1) {
+            // The user and company do not exist (not a migrated client)
+            //     => Redirect them to the welcome page with challenge.
+            else if (
+              associatedCompanies.length < 1 &&
```
Comment (erikataot) [inline frontend/src/common/authentication/LoginRedirect.tsx]:
```suggestion
              associatedCompanies?.length < 1 &&
```

File: frontend/src/features/wizard/components/dashboard/CreateProfileSteps.tsx [hunk 3]
```diff
@@ -193,6 +195,10 @@ export const CreateProfileSteps = React.memo(() => {
         setCompanyLegalName?.(() => companyName);
         setOnRouteBCClientNumber?.(() => clientNumber);
 
+        // Clear any state in migrated client. We no longer need this
+        // once the user has successfully created/claimed their company.
+        setMigratedClient?.(() => undefined);
+
         // Setting the companyId in the sessionStorage so that it can be used
         // used outside of react components;
         sessionStorage.setItem("onRouteBC.user.companyId", companyId);
```
Comment (erikataot) [inline frontend/src/features/wizard/components/dashboard/CreateProfileSteps.tsx:207]: nice

File: frontend/src/features/wizard/pages/CompanyInformationWizardForm.tsx [hunk 4]
```diff
@@ -5,20 +5,44 @@ import "./CompanyInformationWizardForm.scss";
...
+      </Typography>
+      <CustomFormComponent
+        type="input"
+        feature={FEATURE}
+        options={{
+          name: "alternateName",
+          rules: {
+            required: false,
+            validate: {
+              validateAlternateName: (alternateName?: string) =>
```
Comment (erikataot) [inline frontend/src/features/wizard/pages/CompanyInformationWizardForm.tsx]: DEFINITELY NOT suggesting a change here, but ideally this validate logic should go into a helper due to it's "complexity"

**Comment by erikataot** [comment]
This is looking really solid. I am confused though why there's a change in the lock file without a change in the `package.json`.

**Comment by krishnan-aot** [comment]
> This is looking really solid. I am confused though why there's a change in the lock file without a change in the `package.json`.

I executed `npm run audit` because `npm ls --all` kept throwing errors when trying to observe the dependency tree. There have been many dependency updates in the last week.

**Review by erikataot — APPROVED** [review]
nice work!

**Review by zgong-gov — APPROVED** [review]

**Pull request merged**